src/server/routes/api/v1/parking-lot/index.ts

```diff
@@ -0,0 +1,106 @@
+import httpResponse from '../../../../utils/http-response';
+import { UnexistentParkingLotID } from '../../../../error/parking-lot.service';
+import type {
+  FastifyError,
+  FastifyInstance,
+  FastifyRegisterOptions,
+  FastifyReply,
+  FastifyRequest,
+} from 'fastify';
+import validationSchema from './validation-schema';
+
+export default function (
+  fastify: FastifyInstance,
+  _: FastifyRegisterOptions<unknown>,
+  done: (err?: FastifyError) => void,
+): void {
+  fastify.post(
+    '/',
+    { schema: validationSchema.createParkingLotSchema },
+    async (
+      request: FastifyRequest<{
+        Body: {
+          name: string;
+        };
+      }>,
+      reply: FastifyReply,
+    ) => {
+      try {
+        const parkingLot = await fastify.services.parkingLot.create(
+          request.body,
+        );
+
+        return httpResponse.created(reply, parkingLot);
+      } catch (error) {
+        return httpResponse.internalServerError(reply, error);
+      }
+    },
+  );
+
+  fastify.post(
```

Comment:
I think this is wrong, we should be using either `PUT` or `PATCH` but not `POST` (which is intended for creation purposes) here, as the client is probably sending the whole ParkingLot object, we probably want to use `PUT` here.

```suggestion
  fastify.put(
```

Reply:
@Davejs136 we still have `POST` here, we need this to be a `PUT` instead